### PR TITLE
Fix: Problem when duplicating a 'link only' menu item in Mega menu [E…

### PR DESCRIPTION
**Pull request by louiswol94**
…D-9535]

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-elements/assets/js/editor/nested-repeater/controls/repeater.js [hunk 1]
```diff
@@ -24,11 +24,17 @@ export default class Repeater extends elementor.modules.controls.Repeater {
 	}
 
 	onChildviewClickDuplicate( childView ) {
+		const containerModelCid = this.getRepeaterItemContainerModelCidIfExists( childView );
+		const renderAfterInsert = ! containerModelCid ? true : false;
```
Comment (davseve) [inline modules/nested-elements/assets/js/editor/nested-repeater/controls/repeater.js:28]: Don't we always want to render after insert?

**Review by davseve — DISMISSED** [review]

File: assets/dev/js/editor/document/elements/commands/create.js [hunk 2]
```diff
@@ -48,7 +48,13 @@ export class Create extends $e.modules.editor.document.CommandHistoryBase {
 		containers.forEach( ( container ) => {
 			container = container.lookup();
 
-			const createdContainer = container.view.addElement( model, options ).getContainer();
+			const createdElement = container.view.addElement( model, options );
+
+			if ( ! createdElement ) {
```
Comment (udidol) [inline assets/dev/js/editor/document/elements/commands/create.js:53]: I also came across the error here when I was playing with it.
My questions are - does this fix the problem or just handle the error? Isn't an element supposed to be created?

**Review by udidol — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/elements/commands/create.js (hunk 2 above)
Comment (louiswol94) [reply assets/dev/js/editor/document/elements/commands/create.js:53]: @udidol From what I can see this creates the content container - which we don't need.

Comment (louiswol94) [reply assets/dev/js/editor/document/elements/commands/create.js]: @udidol I saw that error came from "creating" a new item, not duplicating. I removed this code as we have a separate task for that, and this code only handled the error. the issue was the `nested-repeater-create-container.js` should not be calling this file in the first place for Link Only items.

**Review by udidol — CHANGES_REQUESTED** [review]

File: modules/nested-elements/assets/js/editor/nested-repeater/controls/repeater.js [hunk 3]
```diff
@@ -24,11 +24,17 @@ export default class Repeater extends elementor.modules.controls.Repeater {
...
 	onChildviewClickDuplicate( childView ) {
+		const containerModelCid = this.getRepeaterItemContainerModelCidIfExists( childView );
+		const renderAfterInsert = ! containerModelCid;
+
 		$e.run( 'document/repeater/duplicate', {
 			container: this.options.container,
 			name: this.model.get( 'name' ),
 			index: childView._index,
-			renderAfterInsert: false,
+			renderAfterInsert,
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply modules/nested-elements/assets/js/editor/nested-repeater/controls/repeater.js]: @udidol Done

**Review by udidol — APPROVED** [review]

**Pull request merged**